### DISCUSSION TITLE
Changing fields in reports that are generated

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Lern/Community#147] [Query] [answered]
Hello @kumarks1122,

There are a few fields in the progress report that I do not require, and I want to remove them
The fields are highlighted in below picture.
![unwantedfields](https://github.com/Sunbird-Lern/Community/assets/65657373/f1b87df1-a697-4e9f-8d25-c0066008a303)

I need to add a field with name of user in this report, how can it be done?
How can I remove or add them during report generation?

cc : @Renuka-S @komalm @Nithin-ns24 @sravani-64 @vinod303 @gandham-santhosh

**Comment by kumarks1122** [comment]
@Xitija We can modify the list of columns from the below code block. 
https://github.com/Sunbird-Lern/data-products/blob/release-5.2.0/lern-data-products/src/main/scala/org/sunbird/lms/exhaust/collection/ProgressExhaustJob.scala#L46-L52
FYI: Getting the name of the user in the progress exhaust is possible. But since the name PII info of the user, it is sensitive data.

**Reply by komalm** [reply]
We want to show the user's first and last name in the same column (it's fine if we can show them in distinct columns). We do not wish to reveal the username.

Could you please advise if this is possible?

**Reply by Krishnaj20** [reply, marked as answer]
@komalm - User's name is considered as PII information. This is currently designed to be shown in the user profile exhaust. Please follow this thread raised by @coolbung where necessary inputs were provided on how to design these new reports.

https://github.com/orgs/Sunbird-Ed/discussions/238

**Reply by Jayaprakash8887** [reply]
@komalm Please let us know if your concern is answered. If yes, kindly mark the thread as answered.
